tests/test_objectives.py

```diff
@@ -23,268 +23,253 @@
+    assert hess_exact[0].shape, (2, 2, 1)
+
+    # Test if these two models have the same call, jacobian, and hessian
+    assert eval_exact[0] == pytest.approx(eval_numerical)
+    assert isinstance(eval_numerical, float)
+    assert isinstance(eval_exact[0][0], float)
+    assert np.squeeze(jac_exact[0], axis=-1) == pytest.approx(jac_numerical)
+    assert isinstance(jac_numerical, np.ndarray)
+    assert np.squeeze(hess_exact[0], axis=-1) == pytest.approx(hess_numerical)
+    assert isinstance(hess_numerical, np.ndarray)
+
+    fit = Fit(chi2_exact, x=xdata, y=ydata, objective=MinimizeModel)
+    fit_exact_result = fit.execute()
+    fit = Fit(model, x=xdata, y=ydata, absolute_sigma=True)
+    fit_num_result = fit.execute()
+    assert fit_exact_result.value(a) == fit_num_result.value(a)
+    assert fit_exact_result.value(b) == fit_num_result.value(b)
+    assert fit_exact_result.stdev(a) == pytest.approx(fit_num_result.stdev(a))
+    assert fit_exact_result.stdev(b) == pytest.approx(fit_num_result.stdev(b))
+
+
+def test_LogLikelihood():
+    """
+    Tests if the LeastSquares objective gives the right shapes of output by
+    comparing with its analytical equivalent.
+    """
+    # TODO: update these tests to use indexed variables in the future
+    a, b = parameters('a, b')
+    i = Idx('i', 100)
+    x, y = variables('x, y')
+    pdf = Exp(x, 1 / a) * Exp(x, b)
+
+    np.random.seed(10)
+    xdata = np.random.exponential(3.5, 100)
+
+    # We use minus loglikelihood for the model, because the objective was
+    # designed to find the maximum when used with a *minimizer*, so it has
+    # opposite sign. Also test MinimizeModel at the same time.
+    logL_model = Model({y: pdf})
+    logL_exact = Model({y: - FlattenSum(log(pdf), i)})
+    logL_numerical = LogLikelihood(logL_model, {x: xdata, y: None})
+    logL_minmodel = MinimizeModel(logL_exact, data={x: xdata, y: None})
+
+    # Test model jacobian and hessian shape
+    eval_exact = logL_exact(x=xdata, a=2, b=3)
+    jac_exact = logL_exact.eval_jacobian(x=xdata, a=2, b=3)
+    hess_exact = logL_exact.eval_hessian(x=xdata, a=2, b=3)
+    eval_minimizemodel = logL_minmodel(a=2, b=3)
+    jac_minimizemodel = logL_minmodel.eval_jacobian(a=2, b=3)
+    hess_minimizemodel = logL_minmodel.eval_hessian(a=2, b=3)
+    eval_numerical = logL_numerical(a=2, b=3)
+    jac_numerical = logL_numerical.eval_jacobian(a=2, b=3)
+    hess_numerical = logL_numerical.eval_hessian(a=2, b=3)
+
+    # TODO: These shapes should not have the ones! This is due to the current
+    # convention that scalars should be returned as a 1d array by Model's.
+    assert eval_exact[0].shape == (1,)
+    assert jac_exact[0].shape == (2, 1)
+    assert hess_exact[0].shape == (2, 2, 1)
+    # Test if identical to MinimizeModel
+    assert eval_exact[0] == pytest.approx(eval_minimizemodel)
+    assert jac_exact[0] == pytest.approx(jac_minimizemodel)
+    assert hess_exact[0] == pytest.approx(hess_minimizemodel)
```

Comment:
Does this do the same thing as the original numpy.testing.assert_almost_equal?

Reply:
`numpy.testing.assert_almost_equal` can compare lists of lists, but when you try to compare with `pytest.approx`, you get an TypeError.